File: Year 2/Algorithms/Data Structures/0.3/main.kt
import java.io.File
import java.io.PrintWriter

fun main()
{
    val fin = File("input.txt")
    val fout = File("output.txt")

    var n = fin.readText().trim().toLong()
    var i = 0L

    val out = PrintWriter(fout)
    while (n != 0L)
    {
        if (n % 2L == 1L)
        {
            out.println(i)
        }
        i++
        n /= 2
    }
    out.close()
}

File: Year 2/Algorithms/Data Structures/0.3/main.cpp
#include <fstream>

int main()
{
    std::ifstream fin("input.txt");
    std::ofstream fout("output.txt");

    long long n, i = 0;
    fin >> n;

    while (n)
    {
        if (n % 2)
        {
            fout << i << "\n";
        }
        i++;
        n /= 2;
    }
    return 0;
}